rewrite-java/src/main/java/org/openrewrite/java/cleanup/HideUtilityClassConstructorVisitor.java

```diff
@@ -0,0 +1,303 @@
+            }
+
+            List<J.MethodDeclaration> methods = c.getBody().getStatements().stream().filter(J.MethodDeclaration.class::isInstance).map(J.MethodDeclaration.class::cast).collect(Collectors.toList());
+            int staticMethodCount = UtilityClassUtilities.countStaticMethods(methods);
+            if (staticMethodCount < 0) {
+                return false;
+            }
+
+            List<J.VariableDeclarations> fields = c.getBody().getStatements().stream().filter(J.VariableDeclarations.class::isInstance).map(J.VariableDeclarations.class::cast).collect(Collectors.toList());
+            int staticFieldCount = UtilityClassUtilities.countStaticFields(fields);
+            if (staticFieldCount < 0) {
+                return false;
+            }
+
+            return staticMethodCount != 0 || staticFieldCount != 0;
+        }
+
+        /**
+         * @return -1 if a non-static field is found, else the count of non-private static fields.
+         */
+        private static int countStaticFields(Collection<J.VariableDeclarations> fields) {
+            int count = 0;
+            for (J.VariableDeclarations field : fields) {
+                if (!field.hasModifier(J.Modifier.Type.Static)) {
+                    return -1;
+                }
+                if (field.hasModifier(J.Modifier.Type.Private)) {
+                    continue;
+                }
+                count++;
+            }
+            return count;
+        }
+
+        /**
+         * @return -1 if a non-static method is found, else the count of non-private static methods.
+         */
+        private static int countStaticMethods(Collection<J.MethodDeclaration> methods) {
+            int count = 0;
+            for (J.MethodDeclaration method : methods) {
+                if (method.isConstructor()) {
+                    continue;
+                }
+                if (!method.hasModifier(J.Modifier.Type.Static)) {
+                    return -1;
+                }
+                if (method.hasModifier(J.Modifier.Type.Private)) {
+                    continue;
+                }
+                count++;
+            }
+            return count;
+        }
+    }
+
+}
+
+@Incubating(since = "7.0.0")
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+@Data
+@With
+class HideUtilityClassConstructorStyle implements JavaStyle {
+    /**
+     * If any of the annotation signatures are present on the utility class, the visitor will ignore operating on the class.
+     * These should be {@link AnnotationMatcher}-compatible, fully-qualified annotation signature strings.
+     */
+    // TODO alt name, `ignorableAnnotations`
+    Collection<String> ignoreIfAnnotatedBy;
+
+    static HideUtilityClassConstructorStyle hideUtilityClassConstructorStyle() {
+        return new HideUtilityClassConstructorStyle(Arrays.asList(
+                "@lombok.experimental.UtilityClass",
+                "@lombok.Data"
+        ));
+    }
+}
```

Comment:
Thoughts welcome on the best way to make "ignorableAnnotations" be configurable-- if at all. It could just be removed for the moment and added later. Also as noted in the issue body, this is just here for the moment until determined whether this should be something configurable at all, a style, or some other config